File: backend/models/Doctor.js
const mongoose = require('mongoose');

const doctorSchema = new mongoose.Schema({
  doctorId: {
    type: Number,
    unique: true
  },
  firstName: {
    type: String,
    required: true,
    trim: true
  },
  lastName: {
    type: String,
    required: true,
    trim: true
  },
  email: {
    type: String,
    required: true,
    unique: true,
    trim: true,
    lowercase: true
  },
  mobileNumber: {
    type: String,
    required: true,
    trim: true
  },
  type: {
    type: String,
    trim: true,
    default: ''
  },
  qualifications: {
    type: String,
    trim: true,
    default: ''
  },
  additionalInfo: {
    type: String,
    trim: true,
    default: ''
  },
  password: {
    type: String,
    required: true
  },
  image: {
    type: String,
    default: ''
  },
  pmdc: {
    type: String,
    trim: true,
    default: ''
  },
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true
});

// Create indexes for better performance
doctorSchema.index({ doctorId: 1 });
doctorSchema.index({ email: 1 });

// Auto-increment doctorId before saving
doctorSchema.pre('save', async function(next) {
  if (this.isNew && !this.doctorId) {
    try {
      // Find the highest existing doctorId and increment it
      const lastDoctor = await this.constructor.findOne({}, {}, { sort: { 'doctorId': -1 } });
      this.doctorId = lastDoctor ? lastDoctor.doctorId + 1 : 1;
    } catch (error) {
      return next(error);
    }
  }
  next();
});

// Simple password storage (no hashing)

// Method to get full name
doctorSchema.virtual('fullName').get(function() {
  return `${this.firstName} ${this.lastName}`;
});

module.exports = mongoose.model('Doctor', doctorSchema);

File: doctor/lib/models/doctor.dart
class Doctor {
  final String id;
  final int doctorId;
  final String firstName;
  final String lastName;
  final String email;
  final String mobileNumber;
  final String? type;
  final String? qualifications;
  final String? additionalInfo;
  final String? imageUrl;

  Doctor({
    required this.id,
    required this.doctorId,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.mobileNumber,
    this.type,
    this.qualifications,
    this.additionalInfo,
    this.imageUrl,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['_id'] as String,
      doctorId: json['doctorId'] as int,
      firstName: json['firstName'] as String,
      lastName: json['lastName'] as String,
      email: json['email'] as String,
      mobileNumber: json['mobileNumber'] as String,
      type: json['type'] as String?,
      qualifications: json['qualifications'] as String?,
      additionalInfo: json['additionalInfo'] as String?,
      imageUrl: json['imageUrl'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'doctorId': doctorId,
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'mobileNumber': mobileNumber,
      'type': type,
      'qualifications': qualifications,
      'additionalInfo': additionalInfo,
      'imageUrl': imageUrl,
    };
  }

  String get fullName => '$firstName $lastName';
}
